packages/fileeditor-extension/src/commands.ts

```diff
@@ -1649,6 +1649,34 @@ export namespace Commands {
     return disposables;
   }
 
+  /**
+   * Add Create New ___ File commands to the FileBrowser menu for common file types associated with available kernels
+   * Returns a DisposableSet containing all added items
+   */
+  export function addKernelLanguageBrowserMenuItems(
+    app: JupyterFrontEnd,
+    availableKernelFileTypes: Iterable<IFileTypeData>
+  ): IDisposable {
+    const disposables = new DisposableSet();
+
+    for (const ext of availableKernelFileTypes) {
+      disposables.add(
+        app.contextMenu.addItem({
+          command: 'filebrowser:create-new-file',
```

Comment:
I'm not super happy with this. Ideally, the filebrowser extension would handle this? 

We'd need to extract `getAvailableKernelFileTypes` and some other APIs out of the `fileeditor-extension`. Do you have thoughts about a better place for this?

Reply:
> But if you do want to move getAvailableKernelFileTypes I would think maybe docregistry would be a good home for it?

Thanks for the suggestion! I made the changes